pkg/services/stats/statsimpl/stats.go

```diff
@@ -1,39 +1,48 @@
-package sqlstore
+package statsimpl
 
 import (
 	"context"
 	"strconv"
 	"time"
 
+	"github.com/grafana/grafana/pkg/infra/db"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/org"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
 	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/services/stats"
 )
 
 const activeUserTimeLimit = time.Hour * 24 * 30
 const dailyActiveUserTimeLimit = time.Hour * 24
 
-func (ss *SQLStore) GetAlertNotifiersUsageStats(ctx context.Context, query *models.GetAlertNotifierUsageStatsQuery) error {
-	return ss.WithDbSession(ctx, func(dbSession *DBSession) error {
-		var rawSQL = `SELECT COUNT(*) AS count, type FROM ` + dialect.Quote("alert_notification") + ` GROUP BY type`
+func ProvideService(db db.DB) stats.Service {
+	return &sqlStatsService{db: db}
+}
+
+type sqlStatsService struct{ db db.DB }
+
+func (ss *sqlStatsService) GetAlertNotifiersUsageStats(ctx context.Context, query *models.GetAlertNotifierUsageStatsQuery) error {
+	return ss.db.WithDbSession(ctx, func(dbSession *db.Session) error {
+		var rawSQL = `SELECT COUNT(*) AS count, type FROM ` + ss.db.GetDialect().Quote("alert_notification") + ` GROUP BY type`
 		query.Result = make([]*models.NotifierUsageStats, 0)
```

Comment:
some candidate future improvements:
- mode also the models under the service (unless it's not possible because of cycle dependencies)
- modify `GetAlertNotifiersUsageStats()` so that returns the stats instead of setting it to query result